src/schema/tokens/pack.ts

```diff
@@ -0,0 +1,97 @@
+import { z } from "zod";
+import { BigNumberishSchema, PriceSchema, RawDateSchema } from "../shared";
+import { NFTInputOrUriSchema } from "./common";
+import {
+  ERC1155WrappableSchema,
+  ERC20WrappableSchema,
+  ERC721WrappableSchema,
+} from "./common/wrap";
+
+/**
+ * @internal
+ */
+const ERC20RewardSchema = ERC20WrappableSchema.omit({
+  quantity: true,
+}).extend({
+  quantityPerReward: PriceSchema,
```

Comment:
wonder if it's really worth it to rename this. would help with code sharing if we didnt

Reply:
I think quantityPerReward is an important distinction here - added it because I originally had just `quantity` as the interface, but when I was using it, it felt like quantity was the number of tokens I was going to wrap (which it isn't), so it felt confusing. What I had in mind was wouldn't want to have a user put something like:
```
quantity: 1
totalRewards: 50
```
expecting to only lose 1 token, and then lose 50, so wanted to make it explicit.